core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -86,9 +86,9 @@ class SharedAllocationHeader {
  public:
   /* Given user memory get pointer to the header */
   KOKKOS_INLINE_FUNCTION static const SharedAllocationHeader* get_header(
-      void* alloc_ptr) {
-    return reinterpret_cast<SharedAllocationHeader*>(
-        reinterpret_cast<char*>(alloc_ptr) - sizeof(SharedAllocationHeader));
```

Comment:
I don't think that this improves this at all. At least reinterpret_cast is a clear sign you do something really naughty, pretending its ok what we are doing by first doing static_cast to void* and then static cast to something else feels just wrong ...

Reply:
I disagree with you but I will revert if need to be.  I am curious how others feel.  @nmm0 @nliber @masterleinad 

The main thing I want is make the argument a pointer to const.

Reply:
I somewhat agree with @crtrott and would prefer
```C++
  return reinterpret_cast<SharedAllocationHeader const*>(
        static_cast<char const*>(alloc_ptr) - sizeof(SharedAllocationHeader));
```

Reply:
See ffcba005d6026002087607933247e7a024257c44